src/main/java/net/patchworkmc/patcher/capabilityinject/CapabilityInjectRewriter.java

```diff
@@ -0,0 +1,162 @@
+package net.patchworkmc.patcher.capabilityinject;
+
+import static org.objectweb.asm.Opcodes.ASM7;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import net.patchworkmc.patcher.Patchwork;
+import net.patchworkmc.patcher.annotation.StringAnnotationHandler;
+import net.patchworkmc.patcher.util.LambdaVisitors;
+
+public class CapabilityInjectRewriter extends ClassVisitor {
+	public static final String CAPABILITY_DESC = "(Lnet/minecraftforge/common/capabilities/Capability;)V";
+	public static final String PREFIX = "patchwork$capabilityInject$";
+
+	private String className;
+	private List<CapabilityInject> injects = new ArrayList<>();
+
+	public CapabilityInjectRewriter(ClassVisitor classVisitor) {
+		super(ASM7, classVisitor);
+	}
+
+	public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+		super.visit(version, access, name, signature, superName, interfaces);
+		this.className = name;
+	}
+
+	public FieldVisitor visitField(int access, String name, String descriptor, String signature, Object value) {
+		FieldVisitor parent = super.visitField(access, name, descriptor, signature, value);
+		return new FieldScanner(parent, access, name, descriptor);
+	}
+
+	public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
```

Comment:
This should check if there are methods starting with the reserved prefix, like EventHandlerRewriter does.